### PR TITLE
Fixes #2950 : Support for browse to command to launch the custom domain URL if present

### LINKED ISSUE #2950: [App Service] [Discussion] Browse command doesn't launch custom domain URL
### Environment summary

**Install Method:** How did you install the CLI? (e.g. pip, interactive script, apt-get, Docker, MSI, nightly)  
Answer here: pip nightly

**CLI Version:** What version of the CLI and modules are installed? (Use `az --version`)  
Answer here:

azure-cli (2.0.3+1.dev20170419)

acr (2.0.1+1.dev20170419)
acs (2.0.3+1.dev20170419)
appservice (0.1.3+1.dev20170419)
batch (2.0.1+1.dev20170419)
cloud (2.0.1+1.dev20170419)
component (2.0.1+1.dev20170419)
configure (2.0.3+1.dev20170419)
container (0.1.3+1.dev20170419)
core (2.0.3+1.dev20170419)
documentdb (0.1.3+1.dev20170419)
feedback (2.0.1+1.dev20170419)
find (0.1.2rc1+1.dev20170312)
iot (0.1.3+1.dev20170419)
keyvault (2.0.1+1.dev20170419)
lab (0.0.2+1.dev20170419)
monitor (0.0.2+1.dev20170419)
network (2.0.3+1.dev20170419)
nspkg (2.0.0+1.dev20170419)
profile (2.0.3+1.dev20170419)
redis (0.2.0+1.dev20170419)
resource (2.0.3+1.dev20170419)
role (2.0.2+1.dev20170419)
sql (2.0.1+1.dev20170419)
storage (2.0.3+1.dev20170419)
vm (2.0.3+1.dev20170419)

Python (Darwin) 2.7.13 (default, Apr  4 2017, 08:47:57) 
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)]

**OS Version:** What OS and version are you using?  
Answer here: macOS

**Shell Type:** What shell are you using? (e.g. bash, cmd.exe, Bash on Windows)  
Answer here: bash

---

### Description

After adding a custom domain, and running `az appservice web browse`, it launches a browser with the `*.azurewebsites.net` URL, as opposed to the custom domain. This obviously isn't a blocking issue, but I wonder if it's the expected behavior. After you add a custom domain, the portal displays that as the main `URL` on the `Overview` blade, and the `Browse` button launches the custom URL as well. For consistency, it would seem ideal for the CLI to echo this behavior.

### PR DESCRIPTION
**Pull request by panchagnula**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -1046,7 +1046,7 @@ def show_container_cd_url(resource_group_name, name, slot=None):
 
 def view_in_browser(resource_group_name, name, slot=None, logs=False):
     site = _generic_site_operation(resource_group_name, name, 'get', slot)
-    url = site.default_host_name
+    url = site.enabled_host_names[0]  # picks the custom domain URL incase a domain is assigned
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Does the webapp service guarantee the first one will be the customized one, rather the default one?  Please confirm the service behavior and add a test for that. The other option is to look for a hostname which is not the default one and also not the scm one.

Comment (panchagnula) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: @yugangw-msft yes the API add the items such that the first item is the domain URL to use.
I have tested this on apps, slots & function apps with custom domain & SSL.

Comment (yugangw-msft) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Okay, let us keep like this then. If it ever gets broken, you own me a test.

**Comment by yugangw-msft** [comment]
To merge after knack conversion.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (panchagnula) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Sure you got it! If this fails you can tell me 'I told you so'

**Comment by tjprescott** [comment]
Due to recent changes to the CLI core, you will need to rebase your changes against the latest in the dev branch.

**Comment by panchagnula** [comment]
@tjprescott rebase to latest dev complete.

**Comment by tjprescott** [comment]
There are two issues you need to resolve to get your PR moving toward green:

Please bump the version number in the setup.py file and HISTORY.RST file for azure-cli-acs and azure-cli-appservice.

```
There are changes to azure-cli-acs and the current version 2.0.23 is already available on PyPI! Bump the version.
Changes are as follows:
* [AKS] fix get-credentials command (#5226)
Error(s) on azure-cli-acs
There are changes to azure-cli-appservice and the current version 0.1.23 is already available on PyPI!Bump the version.
Changes are as follows:
* Fixes #2950 : Support for browse to command to lauch the custom domain URL if present
Error(s) on azure-cli-appservice
The following modules have invalid versions:
azure-cli-acs
azure-cli-appservice
```

Also, is this fixing a bug or does it have a customer-facing change?  If so, it should have an entry in the HISTORY.RST file.

**Comment by derekbekoe** [comment]
Taking a look at the CI failure for AKS. Looks like it's from a recent PR merge.

**Comment by tjprescott** [comment]
Your latest commit "removing the version update for acs module" is failing your build.  Please remove that commit. It should be version 2.0.24.

**Pull request merged**